src/components/app/list/TriggerUrl.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react'
+import { copyToClipboard, Progressing, VisibleModal } from '../../common'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+import { ReactComponent as CopyText } from '../../../assets/icons/ic-copy.svg'
+import Tippy from '@tippyjs/react'
+import { getIngressServiceUrls } from '../service'
+import { KIND } from '../../../config/constants'
+import EmptyState from '../../EmptyState/EmptyState'
+import AppNotDeployed from '../../../assets/img/app-not-deployed.png'
+import { getManifestUrlInfo } from '../../external-apps/ExternalAppService'
+import IndexStore from '../../v2/appDetails/index.store'
+
+interface TriggerURL {
+    appId?: string
+    envId: string
+    installedAppId?: string
+    close: () => void
+    isEAMode?: boolean
+}
+
+export function TriggerUrlModal({ appId, envId, installedAppId, isEAMode, close }: TriggerURL) {
+    const [resp, setResponse] = useState<any>()
+    const [loading, setLoading] = useState(true)
+    const data = { Ingress: [], Service: [] }
+
+    async function getManifest() {
+        try {
+            if (isEAMode) {
+                const response = await getManifestUrlInfo(appId)
+                setResponse(response.result)
+                setLoading(false)
+            } else {
+                const response = await getIngressServiceUrls({ appId, envId, installedAppId })
+                setResponse(response.result)
+                setLoading(false)
+            }
+        } catch (error) {
+            console.log('error')
+        }
+    }
+
+    resp?.map((item) => {
+        if (item?.kind === KIND.SERVICE && item.pointsTo) {
```

Comment:
Why are we adding optional check here? can we use if else here?

Reply:
done